### PR TITLE
Chart fixes for docs snapshots

### PR DESCRIPTION
**Pull request by mattrunyon**
Fixes some issues for charts in the docs snapshot tool. Mainly moving the theme out of ChartUtils so the snapshot tool (in Node) can import ChartUtils and not crash from being unable to handle the css import

### CODE REVIEW

**Comment by mofojed** [comment]
Testing with Enterprise, it looks like the theme is not being loaded:

image

Ran a few commands to test:

```
msftTrades = db.t("LearnDeephaven", "StockTrades").where("Date=`2017-08-25`", "Sym=`MSFT`").view("Sym", "Last", "Size", "ExchangeTimestamp")

timePlot = plot("Microsoft", msftTrades, "ExchangeTimestamp", "Last").yLabel("Last").xLabel("Time").chartTitle("MSFT Chart").show()
```

```
trades = db.t("LearnDeephaven", "StockTrades")
     .where("Date=`2017-08-25`")
     .view("Sym", "Last", "Size", "ExchangeTimestamp")
totalShares = trades.view("Sym", "SharesTraded=Size").sumBy("Sym")
barChart = catPlot("Shares Traded", totalShares, "Sym", "SharesTraded").chartTitle("Total Shares").plotStyle("BAR").show()
pieChart = piePlot("Shares Traded", totalShares, "Sym", "SharesTraded")
     .chartTitle("Total Shares")
     .show()
histChart = histPlot("MSFT", trades.where("Sym=`MSFT`"), "Last",103)
     .chartTitle("Price Intervals")
     .show()
hBarChart = catPlot("Shares Traded", totalShares, "Sym", "SharesTraded").chartTitle("Total Shares").plotStyle("BAR").plotOrientation("horizontal").show()
```

File: packages/chart/src/ChartUtils.js [hunk 1]
```diff
@@ -1206,39 +1217,49 @@ class ChartUtils {
...
       });
     }
 
     return axis;
   }
 
-  static makeDefaultLayout() {
+  static makeDefaultLayout(theme = {}) {
+    let colorway = [];
+    if (theme.colorway) {
```
Comment (mofojed) [inline packages/chart/src/ChartUtils.js:1230]: I'd put this in another static method, `getColorWayFromThemeProperty` or something. Just to keep this method clean

File: packages/chart/src/plotly/Plotly.js [hunk 2]
```diff
@@ -3,14 +3,13 @@
...
 
 // Load in the trace types we need/support
-Plotly.register([
-  require('plotly.js/lib/bar'),
-  require('plotly.js/lib/histogram'),
-  require('plotly.js/lib/pie'),
-  require('plotly.js/lib/ohlc'),
-  require('plotly.js/lib/scattergl'),
-]);
+Plotly.register([bar, histogram, pie, ohlc, scattergl]);
```
Comment (mofojed) [inline packages/chart/src/plotly/Plotly.js:13]: Just curious why change to import?

**Comment by mofojed** [comment]
Nevermind, the Enterprise fix was just using ChartModelFactory instead of creating FigureChartModel directly:

```
 import debounce from 'lodash.debounce';
-import { FigureChartModel } from '@deephaven/chart';
+import { ChartModelFactory, FigureChartModel } from '@deephaven/chart';
 import { ConsoleUtils } from '@deephaven/console';
 import { IrisGridModelFactory } from '@deephaven/iris-grid';
 import Log from '@deephaven/log';
@@ -223,7 +223,9 @@ class ConsolePanel extends PureComponent {
     const id = this.getItemId(name);
     const metadata = { figure: name };
     const makeModel = () =>
-      session.getObject(object).then(result => new FigureChartModel(result));
+      session
+        .getObject(object)
+        .then(result => ChartModelFactory.makeModel(undefined, result));
 
```

File: packages/chart/src/plotly/Plotly.js (hunk 2 above)
Comment (mattrunyon) [reply packages/chart/src/plotly/Plotly.js:13]: It was invalid ESM before. Can't use `require` within an ESM and we're publishing as ESM. Webpack magically makes the old way work

**Review by mofojed — APPROVED** [review]

**Pull request merged**